lib/exprengine.cpp

```diff
@@ -1565,6 +1611,16 @@ bool ExprEngine::BinOpResult::isTrue(ExprEngine::DataBase *dataBase) const
 #endif
 }
 
+bool ExprEngine::BinOpResult::isAlwaysFalse(DataBase *dataBase) const
+{
+    return !isTrue(dataBase);
+}
+
+bool ExprEngine::BinOpResult::isAlwaysTrue(DataBase *dataBase) const
+{
+    return !isAlwaysFalse(dataBase) && !isEqual(dataBase, 0);
```

Comment:
The `!isAlwaysFalse(dataBase)` is just waste of cpu as far as I see here. It calls z3.